SegmentationModule.py

```diff
@@ -0,0 +1,238 @@
+        return mask
+
+    @staticmethod
+    def __get_smooth_mask(sample_mask):
+        """Applies filter to the raw mask to smooth it
+
+        Args:
+            sample_mask: binary mask vector of predictions
+
+        Returns:
+            smooth mask (binary mask vector)"""
+        # Create kernel
+        g_kernel = Gaussian1DKernel(stddev=5)
+
+        # Convolve data
+        smooth_mask = convolve(sample_mask, g_kernel, boundary='extend')
+
+        # to binary mask
+        round_border = 0.3  # if value is higher, it's musical embedding
+        smooth_mask = [int(t > round_border) for t in smooth_mask]
+        return smooth_mask
+
+    @staticmethod
+    def __get_intervals_by_mask(mask):
+        """Translate mask as prediction to the intervals (time codes)
+
+            Args:
+                mask: prediction mask vector
+
+            Returns:
+                the list of time intervals where time interval is the list of the beginning time and
+                the end time. Time stored as indexes of embeddings"""
+        idxs = np.nonzero(mask)[0]  # get indexes of embeddings with music
+
+        max_lag = 4
+        fragments_list = []
+        start_segment = idxs[0]
+        finish_segment = idxs[0]
+        prev_val = idxs[0]
+        for v in idxs:  # unite neighbor mask ones into intervals
+            if v - prev_val < max_lag:
+                prev_val = v
+                finish_segment = v
+            else:
+                if finish_segment - start_segment > 10:
+                    fragments_list.append([start_segment, finish_segment])
+                    start_segment = v
+                    finish_segment = v
+                    prev_val = v
+                else:
+                    start_segment = v
+                    finish_segment = v
+                    prev_val = v
+        if finish_segment - start_segment > 10:
+            fragments_list.append([start_segment, finish_segment])
+        return fragments_list
+
+    @staticmethod
+    def __abs_intervals_to_time(abs_intervals):
+        """Translate mask with indexes of embeddings to time intervals
+
+        Args:
+            abs_intervals: the list of time intervals where time interval is the list of the beginning time and
+                the end time. Time stored as indexes of embeddings
+
+        Returns:
+            the list of time intervals where time interval is the list of the beginning time and the end time. Time
+            stored as string of format 'hh:mm:ss' """
+        sec_to_time = lambda arg: str(datetime.timedelta(seconds=arg))  # callback to translate seconds to string
+        vgg_rescale = 0.96  # 100 embeddings of VGG is 96 seconds
+        # translate embedding-time intervals to time-time intervals
+        intervals = [[sec_to_time(int(vgg_rescale * q)) for q in v] for v in abs_intervals]
+        return intervals
+
+    @staticmethod
+    def __define_callback_list(checkpoint_file):
```

Comment:
I suppose we should have the possibility to use custom callback_list. But for the purpose of the undefined parameter, this default callback_list can be used